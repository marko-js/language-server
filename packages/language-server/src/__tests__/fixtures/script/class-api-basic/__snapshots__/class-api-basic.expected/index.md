## Diagnostics
### Ln 2, Col 1
```marko
  1 | import fancyButton from "<fancy-button>";
> 2 | import { get } from "@ebay/retriever";
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ 'get' is declared but its value is never read.
  3 | /** Hi */
  4 | export interface Input {
  5 |   year: number,
```

### Ln 2, Col 21
```marko
  1 | import fancyButton from "<fancy-button>";
> 2 | import { get } from "@ebay/retriever";
    |                     ^^^^^^^^^^^^^^^^^ Cannot find module '@ebay/retriever' or its corresponding type declarations.
  3 | /** Hi */
  4 | export interface Input {
  5 |   year: number,
```

### Ln 30, Col 14
```marko
  28 |
  29 | static {
> 30 |     function greet() {
     |              ^^^^^ 'greet' is declared but its value is never read.
  31 |         return "hello world";
  32 |     }
  33 | }
```

### Ln 35, Col 12
```marko
  33 | }
  34 |
> 35 | static var useA = true;
     |            ^^^^ 'useA' is declared but its value is never read.
  36 |
  37 | static type Mobile = {
  38 |     brandName: string;
```

### Ln 19, Col 17
```marko
  17 |         console.log("mounted");
  18 |     }
> 19 |     handleClick(ev: MouseEvent) {
     |                 ^^ 'ev' is declared but its value is never read.
  20 |         console.log("clicked");
  21 |     }
  22 | }
```

### Ln 24, Col 9
```marko
  22 | }
  23 |
> 24 | $ const person = {
     |         ^^^^^^ 'person' is declared but its value is never read.
  25 |     name: "Frank",
  26 |     age: 32
  27 | };
```

### Ln 69, Col 13
```marko
  67 |
  68 |     <for|mobile, i, all| of=mobiles>
> 69 |       <span id=i>${mobile} ${i} ${all.length}</span>
     |             ^^ Type 'number' is not assignable to type 'AttrString'.
  70 |     </for>
  71 |
  72 |     <div.container>
```

### Ln 75, Col 18
```marko
  73 |         <${input.renderBody}/>
  74 |     </div>
> 75 |     <fancyButton something=true>${type}</fancyButton>
     |                  ^^^^^^^^^ Argument of type '{ something: boolean; renderBody: () => MarkoReturn<void>; }' is not assignable to parameter of type 'Directives & Input'.
  Object literal may only specify known properties, and '"something"' does not exist in type 'Directives & Input'.
  76 |     <div>Hello $!{"<b>World</b>"}</div>
  77 |     <div>Placeholder example: <code>\${someValue}</code></div>
  78 |
```

### Ln 79, Col 6
```marko
  77 |     <div>Placeholder example: <code>\${someValue}</code></div>
  78 |
> 79 |     <missing/>
     |      ^^^^^^^ Cannot find name 'missing'.
  80 |     <complex-missing/>
  81 | </div>
  82 |
```

