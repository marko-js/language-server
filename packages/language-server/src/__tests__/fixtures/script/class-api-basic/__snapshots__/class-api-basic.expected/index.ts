import FancyButton from "./components/fancy-button/index.marko";
import { get } from "@ebay/retriever";
/** Hi */ export interface Input {
  year: number;
  isSmartOnly: boolean;
  type: string;
  mobileList: Array<Mobile>;
  renderBody: Marko.Body;
}

function greet() {
  return "hello world";
}

var useA = true;
type Mobile = {
  brandName: string;
};
abstract class Component extends Marko.Component<Input> {
  declare state: {
    something: boolean;
  };
  onMount() {
    console.log("mounted");
  }
  handleClick(ev: MouseEvent) {
    console.log("clicked");
  }
}
export { type Component };
(function (this: void) {
  const input = Marko._.any as Input;
  const component = Marko._.any as Component;
  const state = Marko._.state(component);
  const out = Marko._.out;
  const $signal = Marko._.any as AbortSignal;
  const $global = Marko._.getGlobal(
    // @ts-expect-error We expect the compiler to error because we are checking if the MarkoRun.Context is defined.
    (Marko._.error, Marko._.any as MarkoRun.Context),
  );
  Marko._.noop({ component, state, out, input, $global, $signal });
  const person = {
    name: "Frank",
    age: 32,
  };

  const year: number = input.year;
  const isSmartOnly: boolean = input.isSmartOnly;
  const type: string = component.input.type;

  const mobiles: Array<Mobile> = component.input.mobileList;

  mobiles.forEach((mobile: Mobile) => {
    console.log(mobile.brandName);
  });
  Marko._.renderNativeTag("div")()()({
    class: "mobiles__list",
    onClick: component.handleClick,
    ["renderBody" /*div*/]: (() => {
      Marko._.renderNativeTag("p")()()({
        id: "p",
        ["renderBody" /*p*/]: (() => {
          year;
          return () => {
            return Marko._.voidReturn;
          };
        })(),
      });
      if (isSmartOnly) {
        Marko._.renderNativeTag("span")()()({
          class: "subnote",
          ["renderBody" /*span*/]: (() => {
            isSmartOnly;
            return () => {
              return Marko._.voidReturn;
            };
          })(),
        });
      }
      Marko._.renderNativeTag("span")()()({
        class: "subnote",
        ["renderBody" /*span*/]: (() => {
          type;
          return () => {
            return Marko._.voidReturn;
          };
        })(),
      });

      const mobiles: Array<Mobile> = component.input.mobileList;
      mobiles.forEach((mobile: Mobile) => {
        console.log(mobile.brandName);
      });
      Marko._.forOfTag(
        {
          /*for*/ of: mobiles,
        },
        (mobile, i, all) => {
          Marko._.renderNativeTag("span")()()({
            id: i,
            ["renderBody" /*span*/]: (() => {
              mobile;
              i;
              all.length;
              return () => {
                return Marko._.voidReturn;
              };
            })(),
          });
          return Marko._.voidReturn;
        },
      );
      Marko._.renderNativeTag("div")()()({
        class: Marko._.interpolated`container`,
        ["renderBody" /*div*/]: (() => {
          const __marko_internal_tag_1 = input.renderBody;
          Marko._.renderDynamicTag(__marko_internal_tag_1)()()({
            /*input.renderBody*/
          });
          return () => {
            return Marko._.voidReturn;
          };
        })(),
      });
      const __marko_internal_tag_2 = FancyButton;
      Marko._.renderDynamicTag(__marko_internal_tag_2)()()({
        something: true,
        ["renderBody" /*FancyButton*/]: (() => {
          type;
          return () => {
            return Marko._.voidReturn;
          };
        })(),
      });
      Marko._.renderNativeTag("div")()()({
        ["renderBody" /*div*/]: (() => {
          ("<b>World</b>");
          return () => {
            return Marko._.voidReturn;
          };
        })(),
      });
      Marko._.renderNativeTag("div")()()({
        ["renderBody" /*div*/]: (() => {
          Marko._.renderNativeTag("code")()()({
            ["renderBody" /*code*/]: (() => {
              return () => {
                return Marko._.voidReturn;
              };
            })(),
          });
          return () => {
            return Marko._.voidReturn;
          };
        })(),
      });
      const __marko_internal_tag_3 = Missing;
      Marko._.renderDynamicTag(__marko_internal_tag_3)()()({
        /*Missing*/
      });
      const __marko_internal_tag_4 = Marko._.interpolated`complex-missing`;
      Marko._.renderDynamicTag(__marko_internal_tag_4)()()({
        /*complex-missing*/
      });
      return () => {
        return Marko._.voidReturn;
      };
    })(),
  });
  return;
})();
export default new (class Template extends Marko._.Template<{
  render(
    input: Marko.TemplateInput<Input>,
    stream?: {
      write: (chunk: string) => void;
      end: (chunk?: string) => void;
    },
  ): Marko.Out<Component>;

  render(
    input: Marko.TemplateInput<Input>,
    cb?: (err: Error | null, result: Marko.RenderResult<Component>) => void,
  ): Marko.Out<Component>;

  renderSync(input: Marko.TemplateInput<Input>): Marko.RenderResult<Component>;

  renderToString(input: Marko.TemplateInput<Input>): string;

  stream(
    input: Marko.TemplateInput<Input>,
  ): ReadableStream<string> & NodeJS.ReadableStream;

  api: "class";
  _(): () => <__marko_internal_input extends unknown>(
    input: Marko.Directives &
      Input &
      Marko._.Relate<__marko_internal_input, Marko.Directives & Input>,
  ) => Marko._.ReturnWithScope<__marko_internal_input, void>;
}> {})();
