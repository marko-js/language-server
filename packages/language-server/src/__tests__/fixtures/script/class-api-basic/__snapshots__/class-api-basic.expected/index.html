<div data-marko-node-id="0" class="mobiles__list" onClick>
    <p data-marko-node-id="1" id="p">Year: placeholder</p>

    <div>
      <span data-marko-node-id="3" class="subnote">isSmartPhone: placeholder</span>
    </div>

    <span data-marko-node-id="4" class="subnote">phoneType: placeholder</span>
    

    <div>
      <span data-marko-node-id="6" id="dynamic">placeholder placeholder placeholder</span>
    </div>

    <div data-marko-node-id="7">
        
    </div>
    <div>placeholder</div>
    <div data-marko-node-id="10">Hello placeholder</div>
    <div data-marko-node-id="11">Placeholder example: <code data-marko-node-id="12">${someValue}</code></div>

    
    
</div>